### ISSUE TITLE
MVP - Implement the validation endpoint in the Wazuh manager API

### ISSUE THREAD

**Issue by RamosFe** [wazuh/wazuh#24643]
|Objective|
|---|
| #23466 |

## Description

This Issue aims to implement the MVP for the endpoint proposed and initially developed as part of https://github.com/wazuh/wazuh/issues/24369. 

This development includes:
- [x] Complete endpoint development.
- [x] Review the functionality and endpoints that may need modifications due to the changes in the configuration carried out in #24510. (`GET /manager/configuration/{component}/{configuration}` and `GET /cluster/{node_id}/configuration/{component}/{configuration}`) 
- [x] The UTs for the newly added functionality.
- [x] Integration tests for the new endpoint which must cover the following cases:
  - Check the response if the user doesn't have permission to uninstall the agent.
  - Check the response if the user has permission to uninstall the agent:
     - Create a new policy with the `agent:uninstall` action and `*:*:*` resource.
     - Create a role.
     - Assign the newly created policy to the role.
     - Assign the new role to a user.
     - Make a request to the new endpoint with the user.

**Comment by RamosFe** [comment]
## Update - IT

Added Integration Tests for the new `GET /agents/uninstall` endpoint.

### RBAC White Agent Endpoints

Added the following IT to check the response from the endpoint for a user with the `agent:uninstall` policy.

https://github.com/wazuh/wazuh/blob/fe22bf840687ab99d52d521aac90a5060764f803/api/test/integration/test_rbac_white_agent_endpoints.tavern.yaml#L1863-L1879


<details>
<summary>Test Results</summary>


```bash
(env) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh/api/test/integration$ sudo ../../../env/bin/python3 -m pytest test_rbac_black_agent_endpoints.tavern.yaml
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, html-2.1.1, metadata-3.1.1, cov-4.1.0, tavern-1.23.5, anyio-4.1.0, trio-0.8.0
asyncio: mode=auto
collected 42 items                                                                                                                                                                                                                      

test_rbac_black_agent_endpoints.tavern.yaml ..........................................                                                                                                                                            [100%]

=========================================================================================================== warnings summary ============================================================================================================
../../../env/lib/python3.10/site-packages/_pytest/nodes.py:642
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/_pytest/nodes.py:642: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().from_parent(parent=parent, fspath=fspath, path=path, **kw)

test_rbac_black_agent_endpoints.tavern.yaml: 42 warnings
  <frozen importlib._bootstrap>:283: DeprecationWarning: the load_module() method is deprecated and slated for removal in Python 3.12; use exec_module() instead

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================================= 42 passed, 43 warnings in 1055.89s (0:17:35) ==============================================================================================
```

</details>

### RBAC Black Agent Endpoints

Added the following IT to check the response from the endpoint for a user without the `agent:uninstall` policy. 

https://github.com/wazuh/wazuh/blob/fe22bf840687ab99d52d521aac90a5060764f803/api/test/integration/test_rbac_black_agent_endpoints.tavern.yaml#L1683-L1695

<details>
<summary>Test Results</summary>


```bash
(env) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh/api/test/integration$ sudo ../../../env/bin/python3 -m pytest test_rbac_white_agent_endpoints.tavern.yaml
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, html-2.1.1, metadata-3.1.1, cov-4.1.0, tavern-1.23.5, anyio-4.1.0, trio-0.8.0
asyncio: mode=auto
collected 42 items                                                                                                                                                                                                                      

test_rbac_white_agent_endpoints.tavern.yaml ..........................................                                                                                                                                            [100%]

=========================================================================================================== warnings summary ============================================================================================================
../../../env/lib/python3.10/site-packages/_pytest/nodes.py:642
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/_pytest/nodes.py:642: PytestRemovedIn8Warning: The (fspath: py.path.local) argument to YamlFile is deprecated. Please use the (path: pathlib.Path) argument instead.
  See https://docs.pytest.org/en/latest/deprecations.html#fspath-argument-for-node-constructors-replaced-with-pathlib-path
    return super().from_parent(parent=parent, fspath=fspath, path=path, **kw)

test_rbac_white_agent_endpoints.tavern.yaml: 42 warnings
  <frozen importlib._bootstrap>:283: DeprecationWarning: the load_module() method is deprecated and slated for removal in Python 3.12; use exec_module() instead

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================================= 42 passed, 43 warnings in 1118.41s (0:18:38) ==============================================================================================
```

</details>

**Comment by RamosFe** [comment]
## Update - UT

Added new UT for the functions related to the new endpoint `GET /agents/uninstall`

<details>
<summary>Controllers test results</summary>

```
(env) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh$ PYTHONPATH=/home/federamos/Documents/Wazuh/Repositories/wazuh/framework/:/home/federamos/Documents/Wazuh/Repositories/wazuh/api   pytest api/api/controllers/
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/api/api
configfile: pytest.ini
plugins: asyncio-0.18.1, html-2.1.1, metadata-3.1.1, cov-4.1.0, tavern-1.23.5, anyio-4.1.0, trio-0.8.0
asyncio: mode=auto
collected 210 items                                                                                                                                                                                                                     

api/api/controllers/test/test_active_response_controller.py .                                                                                                                                                                     [  0%]
api/api/controllers/test/test_agent_controller.py ............................................                                                                                                                                    [ 21%]
api/api/controllers/test/test_cdb_list_controller.py ......                                                                                                                                                                       [ 24%]
api/api/controllers/test/test_ciscat_controller.py .                                                                                                                                                                              [ 24%]
api/api/controllers/test/test_cluster_controller.py ........................                                                                                                                                                      [ 36%]
api/api/controllers/test/test_controller_util.py ..                                                                                                                                                                               [ 37%]
api/api/controllers/test/test_decoder_controller.py .......                                                                                                                                                                       [ 40%]
api/api/controllers/test/test_default_controller.py .                                                                                                                                                                             [ 40%]
api/api/controllers/test/test_event_controller.py .                                                                                                                                                                               [ 41%]
api/api/controllers/test/test_experimental_controller.py ...............                                                                                                                                                          [ 48%]
api/api/controllers/test/test_manager_controller.py .....................                                                                                                                                                         [ 58%]
api/api/controllers/test/test_mitre_controller.py .......                                                                                                                                                                         [ 61%]
api/api/controllers/test/test_overview_controller.py .                                                                                                                                                                            [ 62%]
api/api/controllers/test/test_rootcheck_controller.py ....                                                                                                                                                                        [ 64%]
api/api/controllers/test/test_rule_controller.py ........                                                                                                                                                                         [ 68%]
api/api/controllers/test/test_sca_controller.py ..                                                                                                                                                                                [ 69%]
api/api/controllers/test/test_security_controller.py ...................................................                                                                                                                          [ 93%]
api/api/controllers/test/test_syscheck_controller.py ....                                                                                                                                                                         [ 95%]
api/api/controllers/test/test_syscollector_controller.py .........                                                                                                                                                                [ 99%]
api/api/controllers/test/test_task_controller.py .                                                                                                                                                                                [100%]

=========================================================================================================== warnings summary ============================================================================================================
env/lib/python3.10/site-packages/connexion/json_schema.py:16
env/lib/python3.10/site-packages/connexion/json_schema.py:16
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/connexion/json_schema.py:16: DeprecationWarning: jsonschema.RefResolver is deprecated as of v4.18.0, in favor of the https://github.com/python-jsonschema/referencing library, which provides more compliant referencing behavior as well as more flexible APIs for customization. A future release will remove RefResolver. Please file a feature request (on referencing) if you are missing an API for the kind of customization you need.
    from jsonschema import Draft4Validator, RefResolver

env/lib/python3.10/site-packages/connexion/json_schema.py:17
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/connexion/json_schema.py:17: DeprecationWarning: jsonschema.exceptions.RefResolutionError is deprecated as of version 4.18.0. If you wish to catch potential reference resolution errors, directly catch referencing.exceptions.Unresolvable.
    from jsonschema.exceptions import RefResolutionError, ValidationError  # noqa

env/lib/python3.10/site-packages/connexion/validators/form_data.py:4
env/lib/python3.10/site-packages/connexion/validators/form_data.py:4
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/connexion/validators/form_data.py:4: DeprecationWarning: Accessing jsonschema.draft4_format_checker is deprecated and will be removed in a future release. Instead, use the FORMAT_CHECKER attribute on the corresponding Validator.
    from jsonschema import ValidationError, draft4_format_checker

env/lib/python3.10/site-packages/connexion/validators/json.py:6
env/lib/python3.10/site-packages/connexion/validators/json.py:6
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/connexion/validators/json.py:6: DeprecationWarning: Accessing jsonschema.draft4_format_checker is deprecated and will be removed in a future release. Instead, use the FORMAT_CHECKER attribute on the corresponding Validator.
    from jsonschema import Draft4Validator, ValidationError, draft4_format_checker

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================================================================== 210 passed, 7 warnings in 1.47s ====================================================================================================
```

</details>


<details>
<summary>Framework test results</summary>

```
(env) federamos@pop-os:~/Documents/Wazuh/Repositories/wazuh$ sudo PYTHONPATH=/home/federamos/Documents/Wazuh/Repositories/wazuh/framework/:/home/federamos/Documents/Wazuh/Repositories/wazuh/api  env/bin/python3 -m pytest ./framework/wazuh/tests/test_agent.py
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/federamos/Documents/Wazuh/Repositories/wazuh/framework
configfile: pytest.ini
plugins: asyncio-0.18.1, html-2.1.1, metadata-3.1.1, cov-4.1.0, tavern-1.23.5, anyio-4.1.0, trio-0.8.0
asyncio: mode=auto
collected 130 items                                                                                                                                                                                                                     

framework/wazuh/tests/test_agent.py ..................................................................................................................................                                                            [100%]

=========================================================================================================== warnings summary ============================================================================================================
env/lib/python3.10/site-packages/connexion/json_schema.py:16
env/lib/python3.10/site-packages/connexion/json_schema.py:16
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/connexion/json_schema.py:16: DeprecationWarning: jsonschema.RefResolver is deprecated as of v4.18.0, in favor of the https://github.com/python-jsonschema/referencing library, which provides more compliant referencing behavior as well as more flexible APIs for customization. A future release will remove RefResolver. Please file a feature request (on referencing) if you are missing an API for the kind of customization you need.
    from jsonschema import Draft4Validator, RefResolver

env/lib/python3.10/site-packages/connexion/json_schema.py:17
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/connexion/json_schema.py:17: DeprecationWarning: jsonschema.exceptions.RefResolutionError is deprecated as of version 4.18.0. If you wish to catch potential reference resolution errors, directly catch referencing.exceptions.Unresolvable.
    from jsonschema.exceptions import RefResolutionError, ValidationError  # noqa

env/lib/python3.10/site-packages/connexion/validators/form_data.py:4
env/lib/python3.10/site-packages/connexion/validators/form_data.py:4
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/connexion/validators/form_data.py:4: DeprecationWarning: Accessing jsonschema.draft4_format_checker is deprecated and will be removed in a future release. Instead, use the FORMAT_CHECKER attribute on the corresponding Validator.
    from jsonschema import ValidationError, draft4_format_checker

env/lib/python3.10/site-packages/connexion/validators/json.py:6
env/lib/python3.10/site-packages/connexion/validators/json.py:6
  /home/federamos/Documents/Wazuh/Repositories/wazuh/env/lib/python3.10/site-packages/connexion/validators/json.py:6: DeprecationWarning: Accessing jsonschema.draft4_format_checker is deprecated and will be removed in a future release. Instead, use the FORMAT_CHECKER attribute on the corresponding Validator.
    from jsonschema import Draft4Validator, ValidationError, draft4_format_checker

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================================================================== 130 passed, 7 warnings in 3.83s ====================================================================================================
```

</details>

**Comment by RamosFe** [comment]
## Update - Anti-Tampering configuration section

After testing the new configuration and requesting the configuration from `GET /manager/configuration/{component}/{configuration}` and `GET /cluster/{node_id}/configuration/{component}/{configuration}`, the new section doesn't appear in the response:

<details>
<summary>Configuration</summary>

```xml
  <anti_tampering>
    <package_uninstallation>yes</package_uninstallation>
  </anti_tampering>
```

</details>

<details>
<summary>Response</summary>

```json
{
    "data": {
        "affected_items": [
            {
                "global": {
                    "jsonout_output": "yes",
                    "alerts_log": "yes",
                    "logall": "no",
                    "logall_json": "no",
                    "email_notification": "no",
                    "smtp_server": "smtp.example.wazuh.com",
                    "email_from": "wazuh@example.wazuh.com",
                    "email_to": "recipient@example.wazuh.com",
                    "email_maxperhour": "12",
                    "email_log_source": "alerts.log",
                    "agents_disconnection_time": "10m",
                    "agents_disconnection_alert_time": "0",
                    "update_check": "yes",
                    "white_list": [
                        "127.0.0.1",
                        "^localhost.localdomain$",
                        "192.168.1.1"
                    ]
                },
                "alerts": {
                    "log_alert_level": "3",
                    "email_alert_level": "12"
                },
                "remote": [
                    {
                        "connection": "secure",
                        "port": "1514",
                        "protocol": [
                            "tcp"
                        ],
                        "queue_size": "131072"
                    }
                ],
                "rootcheck": {
                    "disabled": "no",
                    "check_files": "yes",
                    "check_trojans": "yes",
                    "check_dev": "yes",
                    "check_sys": "yes",
                    "check_pids": "yes",
                    "check_ports": "yes",
                    "check_if": "yes",
                    "frequency": "43200",
                    "rootkit_files": [
                        "etc/rootcheck/rootkit_files.txt"
                    ],
                    "rootkit_trojans": [
                        "etc/rootcheck/rootkit_trojans.txt"
                    ],
                    "skip_nfs": "yes",
                    "ignore": "/var/lib/docker/overlay2"
                },
                "cis-cat": {
                    "disabled": "yes",
                    "timeout": "1800",
                    "interval": "1d",
                    "scan-on-start": "yes",
                    "java_path": "wodles/java",
                    "ciscat_path": "wodles/ciscat"
                },
                "osquery": {
                    "disabled": "yes",
                    "run_daemon": "yes",
                    "log_path": "/var/log/osquery/osqueryd.results.log",
                    "config_path": "/etc/osquery/osquery.conf",
                    "add_labels": "yes"
                },
                "syscollector": {
                    "disabled": "no",
                    "interval": "1h",
                    "scan_on_start": "yes",
                    "hardware": "yes",
                    "os": "yes",
                    "network": "yes",
                    "packages": "yes",
                    "ports": {
                        "all": "no",
                        "item": "yes"
                    },
                    "processes": "yes",
                    "synchronization": {
                        "max_eps": [
                            "10"
                        ]
                    }
                },
                "sca": {
                    "enabled": "yes",
                    "scan_on_start": "yes",
                    "interval": "12h",
                    "skip_nfs": "yes"
                },
                "vulnerability-detection": {
                    "enabled": "yes",
                    "index-status": "yes",
                    "feed-update-interval": "60m"
                },
                "indexer": {
                    "enabled": "yes",
                    "hosts": [
                        "https://0.0.0.0:9200"
                    ],
                    "ssl": {
                        "certificate_authorities": [
                            {
                                "ca": [
                                    "/etc/filebeat/certs/root-ca.pem"
                                ]
                            }
                        ],
                        "certificate": [
                            "/etc/filebeat/certs/filebeat.pem"
                        ],
                        "key": [
                            "/etc/filebeat/certs/filebeat-key.pem"
                        ]
                    }
                },
                "syscheck": {
                    "disabled": "no",
                    "frequency": "43200",
                    "scan_on_start": "yes",
                    "alert_new_files": "yes",
                    "auto_ignore": {
                        "frequency": "10",
                        "timeframe": "3600",
                        "item": "no"
                    },
                    "directories": [
                        {
                            "path": "/etc"
                        },
                        {
                            "path": "/usr/bin"
                        },
                        {
                            "path": "/usr/sbin"
                        },
                        {
                            "path": "/bin"
                        },
                        {
                            "path": "/sbin"
                        },
                        {
                            "path": "/boot"
                        }
                    ],
                    "ignore": [
                        "/etc/mtab",
                        "/etc/hosts.deny",
                        "/etc/mail/statistics",
                        "/etc/random-seed",
                        "/etc/random.seed",
                        "/etc/adjtime",
                        "/etc/httpd/logs",
                        "/etc/utmpx",
                        "/etc/wtmpx",
                        "/etc/cups/certs",
                        "/etc/dumpdates",
                        "/etc/svc/volatile",
                        {
                            "type": "sregex",
                            "item": ".log$|.swp$"
                        }
                    ],
                    "nodiff": [
                        "/etc/ssl/private.key"
                    ],
                    "skip_nfs": "yes",
                    "skip_dev": "yes",
                    "skip_proc": "yes",
                    "skip_sys": "yes",
                    "process_priority": "10",
                    "max_eps": "50",
                    "synchronization": {
                        "enabled": "yes",
                        "interval": "5m",
                        "max_eps": "10"
                    }
                },
                "command": [
                    {
                        "name": "disable-account",
                        "executable": "disable-account",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "restart-wazuh",
                        "executable": "restart-wazuh"
                    },
                    {
                        "name": "firewall-drop",
                        "executable": "firewall-drop",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "host-deny",
                        "executable": "host-deny",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "route-null",
                        "executable": "route-null",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "win_route-null",
                        "executable": "route-null.exe",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "netsh",
                        "executable": "netsh.exe",
                        "timeout_allowed": "yes"
                    }
                ],
                "localfile": [
                    {
                        "log_format": "syslog",
                        "location": "/var/ossec/logs/active-responses.log"
                    },
                    {
                        "log_format": "syslog",
                        "location": "/var/log/dpkg.log"
                    },
                    {
                        "log_format": "command",
                        "command": "df -P",
                        "frequency": "360"
                    },
                    {
                        "log_format": "full_command",
                        "command": "netstat -tulpn | sed 's/\\([[:alnum:]]\\+\\)\\ \\+[[:digit:]]\\+\\ \\+[[:digit:]]\\+\\ \\+\\(.*\\):\\([[:digit:]]*\\)\\ \\+\\([0-9\\.\\:\\*]\\+\\).\\+\\ \\([[:digit:]]*\/[[:alnum:]\\-]*\\).*/\\1 \\2 == \\3 == \\4 \\5/' | sort -k 4 -g | sed 's/ == \\(.*\\) ==/:\\1/' | sed 1,2d",
                        "alias": "netstat listening ports",
                        "frequency": "360"
                    },
                    {
                        "log_format": "full_command",
                        "command": "last -n 20",
                        "frequency": "360"
                    }
                ],
                "ruleset": {
                    "decoder_dir": [
                        "ruleset/decoders",
                        "etc/decoders"
                    ],
                    "rule_dir": [
                        "ruleset/rules",
                        "etc/rules"
                    ],
                    "rule_exclude": [
                        "0215-policy_rules.xml"
                    ],
                    "list": [
                        "etc/lists/audit-keys",
                        "etc/lists/amazon/aws-eventnames",
                        "etc/lists/security-eventchannel"
                    ]
                },
                "auth": {
                    "disabled": "no",
                    "port": "1515",
                    "use_source_ip": "no",
                    "purge": "yes",
                    "use_password": "no",
                    "ciphers": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH",
                    "ssl_verify_host": "no",
                    "ssl_manager_cert": "etc/sslmanager.cert",
                    "ssl_manager_key": "etc/sslmanager.key",
                    "ssl_auto_negotiate": "no"
                },
                "cluster": {
                    "name": "wazuh",
                    "node_name": "master-node",
                    "node_type": "master",
                    "key": "9d273b53510fef702b54a92e9cffc82e",
                    "port": "1516",
                    "bind_addr": "0.0.0.0",
                    "nodes": [
                        "wazuh-master"
                    ],
                    "hidden": "no",
                    "disabled": "no"
                }
            }
        ],
        "total_affected_items": 1,
        "total_failed_items": 0,
        "failed_items": []
    },
    "message": "Configuration was successfully read in specified node",
    "error": 0
}


```

</details>

### Changes

We need to make the following changes:
- [ ] Add the new section in `configuration.py` to read it:

https://github.com/wazuh/wazuh/blob/d0395507c95bc0c534073ab488bd9f0aa37b68be/framework/wazuh/core/configuration.py#L35-L114

We need to add the following entry:

```python
'anti-tampering': {
  'type': 'last',
  'list_options: []
}
```

- [ ] Add the new component and configuration in the validations:

The two endpoints mentioned in the issue have the parameters `component` and `configuration`, used for retrieving specific parts of the configuration. We need to add the new component and configuration to avoid validation errors:

https://github.com/wazuh/wazuh/blob/d0395507c95bc0c534073ab488bd9f0aa37b68be/api/api/validator.py#L243-L260

- [ ] Add the new component and configuration in the `spec.yaml`:

Finally we need to add the component and configuration described above in the `spec.yaml` list for the two endpoints.

## Working branch

Due to the nature of. `ossec.conf`, that checks if the section of the configuration actually exists, we should add this changes in a branch that uses the branch from this PR as base https://github.com/wazuh/wazuh/pull/24526. To avoid having 2 PR associated to 1 Issue, I suggest we create a new Issue for this changes. This allow as to mantain the 1 to 1 consistency of PR/Issue and it allows unblocking the development of the script that is waiting for the new endpoint.

I will discuss it with the team to check the best approach.

CC: @fdalmaup

**Comment by RamosFe** [comment]
## Update - `On-Hold`

This Issue is moved to `on-hold` to prioritize the Release Testing for `4.9.0` Alpha 3 (Issue: https://github.com/wazuh/wazuh/issues/24714)

## Working Branch - Conclusion

After consulting it with the team, we are going to use `enhancement/24510-uninstall-validation` as the base branch of the Issue.

**Comment by RamosFe** [comment]
## Update - Configuration

After making the changes described in https://github.com/wazuh/wazuh/issues/24643#issuecomment-2243865973. The following results were obtained


<details>
<summary>Obtaining configuration :green_circle: </summary>

Request:
```http
GET /cluster/000/configuration?pretty=false&wait_for_complete=false
```
> The results are the same for `GET /manager/configuration`.

Response
```json
{
    "data": {
        "affected_items": [
            {
                "global": {
                    "jsonout_output": "yes",
                    "alerts_log": "yes",
                    "logall": "no",
                    "logall_json": "no",
                    "email_notification": "no",
                    "smtp_server": "smtp.example.wazuh.com",
                    "email_from": "wazuh@example.wazuh.com",
                    "email_to": "recipient@example.wazuh.com",
                    "email_maxperhour": "12",
                    "email_log_source": "alerts.log",
                    "agents_disconnection_time": "10m",
                    "agents_disconnection_alert_time": "0",
                    "update_check": "yes",
                    "white_list": [
                        "127.0.0.1",
                        "^localhost.localdomain$",
                        "192.168.1.1"
                    ]
                },
                "alerts": {
                    "log_alert_level": "3",
                    "email_alert_level": "12"
                },
                "remote": [
                    {
                        "connection": "secure",
                        "port": "1514",
                        "protocol": [
                            "tcp"
                        ],
                        "queue_size": "131072"
                    }
                ],
                "rootcheck": {
                    "disabled": "no",
                    "check_files": "yes",
                    "check_trojans": "yes",
                    "check_dev": "yes",
                    "check_sys": "yes",
                    "check_pids": "yes",
                    "check_ports": "yes",
                    "check_if": "yes",
                    "frequency": "43200",
                    "rootkit_files": [
                        "etc/rootcheck/rootkit_files.txt"
                    ],
                    "rootkit_trojans": [
                        "etc/rootcheck/rootkit_trojans.txt"
                    ],
                    "skip_nfs": "yes",
                    "ignore": "/var/lib/docker/overlay2"
                },
                "cis-cat": {
                    "disabled": "yes",
                    "timeout": "1800",
                    "interval": "1d",
                    "scan-on-start": "yes",
                    "java_path": "wodles/java",
                    "ciscat_path": "wodles/ciscat"
                },
                "osquery": {
                    "disabled": "yes",
                    "run_daemon": "yes",
                    "log_path": "/var/log/osquery/osqueryd.results.log",
                    "config_path": "/etc/osquery/osquery.conf",
                    "add_labels": "yes"
                },
                "syscollector": {
                    "disabled": "no",
                    "interval": "1h",
                    "scan_on_start": "yes",
                    "hardware": "yes",
                    "os": "yes",
                    "network": "yes",
                    "packages": "yes",
                    "ports": {
                        "all": "no",
                        "item": "yes"
                    },
                    "processes": "yes",
                    "synchronization": {
                        "max_eps": [
                            "10"
                        ]
                    }
                },
                "sca": {
                    "enabled": "yes",
                    "scan_on_start": "yes",
                    "interval": "12h",
                    "skip_nfs": "yes"
                },
                "vulnerability-detection": {
                    "enabled": "yes",
                    "index-status": "yes",
                    "feed-update-interval": "60m"
                },
                "indexer": {
                    "enabled": "yes",
                    "hosts": [
                        "https://0.0.0.0:9200"
                    ],
                    "ssl": {
                        "certificate_authorities": [
                            {
                                "ca": [
                                    "/etc/filebeat/certs/root-ca.pem"
                                ]
                            }
                        ],
                        "certificate": [
                            "/etc/filebeat/certs/filebeat.pem"
                        ],
                        "key": [
                            "/etc/filebeat/certs/filebeat-key.pem"
                        ]
                    }
                },
                "syscheck": {
                    "disabled": "no",
                    "frequency": "43200",
                    "scan_on_start": "yes",
                    "alert_new_files": "yes",
                    "auto_ignore": {
                        "frequency": "10",
                        "timeframe": "3600",
                        "item": "no"
                    },
                    "directories": [
                        {
                            "path": "/etc"
                        },
                        {
                            "path": "/usr/bin"
                        },
                        {
                            "path": "/usr/sbin"
                        },
                        {
                            "path": "/bin"
                        },
                        {
                            "path": "/sbin"
                        },
                        {
                            "path": "/boot"
                        }
                    ],
                    "ignore": [
                        "/etc/mtab",
                        "/etc/hosts.deny",
                        "/etc/mail/statistics",
                        "/etc/random-seed",
                        "/etc/random.seed",
                        "/etc/adjtime",
                        "/etc/httpd/logs",
                        "/etc/utmpx",
                        "/etc/wtmpx",
                        "/etc/cups/certs",
                        "/etc/dumpdates",
                        "/etc/svc/volatile",
                        {
                            "type": "sregex",
                            "item": ".log$|.swp$"
                        }
                    ],
                    "nodiff": [
                        "/etc/ssl/private.key"
                    ],
                    "skip_nfs": "yes",
                    "skip_dev": "yes",
                    "skip_proc": "yes",
                    "skip_sys": "yes",
                    "process_priority": "10",
                    "max_eps": "50",
                    "synchronization": {
                        "enabled": "yes",
                        "interval": "5m",
                        "max_eps": "10"
                    }
                },
                "command": [
                    {
                        "name": "disable-account",
                        "executable": "disable-account",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "restart-wazuh",
                        "executable": "restart-wazuh"
                    },
                    {
                        "name": "firewall-drop",
                        "executable": "firewall-drop",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "host-deny",
                        "executable": "host-deny",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "route-null",
                        "executable": "route-null",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "win_route-null",
                        "executable": "route-null.exe",
                        "timeout_allowed": "yes"
                    },
                    {
                        "name": "netsh",
                        "executable": "netsh.exe",
                        "timeout_allowed": "yes"
                    }
                ],
                "localfile": [
                    {
                        "log_format": "syslog",
                        "location": "/var/ossec/logs/active-responses.log"
                    },
                    {
                        "log_format": "syslog",
                        "location": "/var/log/dpkg.log"
                    },
                    {
                        "log_format": "command",
                        "command": "df -P",
                        "frequency": "360"
                    },
                    {
                        "log_format": "full_command",
                        "command": "netstat -tulpn | sed 's/\\([[:alnum:]]\\+\\)\\ \\+[[:digit:]]\\+\\ \\+[[:digit:]]\\+\\ \\+\\(.*\\):\\([[:digit:]]*\\)\\ \\+\\([0-9\\.\\:\\*]\\+\\).\\+\\ \\([[:digit:]]*\/[[:alnum:]\\-]*\\).*/\\1 \\2 == \\3 == \\4 \\5/' | sort -k 4 -g | sed 's/ == \\(.*\\) ==/:\\1/' | sed 1,2d",
                        "alias": "netstat listening ports",
                        "frequency": "360"
                    },
                    {
                        "log_format": "full_command",
                        "command": "last -n 20",
                        "frequency": "360"
                    }
                ],
                "ruleset": {
                    "decoder_dir": [
                        "ruleset/decoders",
                        "etc/decoders"
                    ],
                    "rule_dir": [
                        "ruleset/rules",
                        "etc/rules"
                    ],
                    "rule_exclude": [
                        "0215-policy_rules.xml"
                    ],
                    "list": [
                        "etc/lists/audit-keys",
                        "etc/lists/amazon/aws-eventnames",
                        "etc/lists/security-eventchannel"
                    ]
                },
                "auth": {
                    "disabled": "no",
                    "port": "1515",
                    "use_source_ip": "no",
                    "purge": "yes",
                    "use_password": "no",
                    "ciphers": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH",
                    "ssl_verify_host": "no",
                    "ssl_manager_cert": "etc/sslmanager.cert",
                    "ssl_manager_key": "etc/sslmanager.key",
                    "ssl_auto_negotiate": "no"
                },
                "cluster": {
                    "name": "wazuh",
                    "node_name": "master-node",
                    "node_type": "master",
                    "key": "9d273b53510fef702b54a92e9cffc82e",
                    "port": "1516",
                    "bind_addr": "0.0.0.0",
                    "nodes": [
                        "wazuh-master"
                    ],
                    "hidden": "no",
                    "disabled": "no"
                },
                "anti_tampering": {
                    "package_uninstallation": "yes"
                }
            }
        ],
        "total_affected_items": 1,
        "total_failed_items": 0,
        "failed_items": []
    },
    "message": "Configuration was successfully read in specified node",
    "error": 0
}
```

</details>



<details>
<summary>Obtaining active configuration :red_circle: </summary>

Request:
```http
GET /manager/configuration/agent/anti_tampering?pretty=false&wait_for_complete=false
```

> Same response for GET /cluster/{node_id}/configuration/agent/anti_tampering

Response:
```json
{
    "data": {
        "affected_items": [],
        "total_affected_items": 0,
        "total_failed_items": 1,
        "failed_items": [
            {
                "error": {
                    "code": 1116,
                    "message": "Requested component configuration does not exist: agent:anti_tampering",
                    "remediation": "Please, visit the official documentation (https://documentation.wazuh.com/4.10/user-manual/api/reference.html#operation/api.controllers.agents_controller.get_agent_config) to check available component configurations"
                },
                "id": [
                    "master-node"
                ]
            }
        ]
    },
    "message": "Could not read active configuration in specified node",
    "error": 1
}
```

</details>


<details>
<summary>Agent active configuration :green_circle: </summary>

Request:
```http
GET /agents/:agent_id/config/agent/anti_tampering?pretty=false&wait_for_complete=false
```

Response:
```json
{
    "data": {
        "package_uninstallation": [
            "yes"
        ]
    },
    "error": 0
}
```



</details>


### Error reasons

When obtaining the `agent` active configuration the following function is used:

https://github.com/wazuh/wazuh/blob/ef3a8bcd10656f5c8e37d1e9f99bd4943e3a99cc/framework/wazuh/core/configuration.py#L1257-L1285

This uses the `remote` socket to get the result. The changes made in https://github.com/wazuh/wazuh/pull/24526 added the new `configuration=anti_tampering`. 

https://github.com/wazuh/wazuh/blob/c0f794665de9ff8eb017073a07da6eb9d8c914a2/src/client-agent/agcom.c#L95-L106

In the case of the manager, the socket `analysis` is used, as shown in the following code:

https://github.com/wazuh/wazuh/blob/ef3a8bcd10656f5c8e37d1e9f99bd4943e3a99cc/framework/wazuh/core/configuration.py#L1174-L1210

### Conclusion

We should check with the @wazuh/devel-agent team if this is the expected behavior and the new configuration shouldn't be added in the `ossec.conf` of the master or if we have to make some changes in the `analysis` socket when the message `getconfig anti_tampering` is sent.

**Comment by RamosFe** [comment]
## Update

After talking with the team, the configuration is only used for the agents, so the error shown in the comment above is the expected behaviour. No changes must be made to `GET /manager/configuration/{component}/{configuration} ` and `GET /cluster/{node_id}/configuration/{component}/{configuration}`.
